stdlib/src/builtin/bin.mojo

```diff
@@ -0,0 +1,81 @@
+# ===----------------------------------------------------------------------=== #
```

Comment:
Once `Indexer` is delivered, this module will be much simpler, potentially just a single function. It might make sense to move it under `builtin/int`. Let's see when that happens.

Reply:
My thought process was that the main implementation needs to be `bin[type: DType](num: Scalar[type])` in case something like `DType.int128` is added later on which could be larger than `Int`. In which case we need exactly two functions.

Reply:
On that note, should we just generalize `bin` to accept SIMD sizes larger than one?

Reply:
I would be happy to do that, should that produce a list representation such as `"[0b1, 0b0]"`?

Reply:
Why not just one long string?

Reply:
Oh I see, leading zeros and signs would be a problem. Yeah, let's table this, I didn't think through. Maybe we can circle back to it later.

Reply:
Could be a bit awkward to read, but one long string would be easier to parse back into a `SIMD` if that was something we wanted to do.